python/ray/train/callbacks/logging.py

```diff
@@ -129,55 +118,13 @@ def start_training(self, logdir: str, **info):
             json.dump([], f, cls=SafeFallbackEncoder)
 
     def handle_result(self, results: List[Dict], **info):
-        if self._workers_to_log is None or results is None:
-            results_to_log = results
-        else:
-            results_to_log = [
-                result for i, result in enumerate(results)
-                if i in self._workers_to_log
-            ]
         with open(self._log_path, "r+") as f:
             loaded_results = json.load(f)
             f.seek(0)
-            json.dump(
-                loaded_results + [results_to_log], f, cls=SafeFallbackEncoder)
-
-
-class TrainingSingleWorkerLoggingCallback(TrainingLogdirCallback, abc.ABC):
-    """Abstract Train logging callback class.
-
-    Allows only for single-worker logging.
+            json.dump(loaded_results + [results], f, cls=SafeFallbackEncoder)
 
-    Args:
-        logdir (Optional[str]): Path to directory where the results file
-            should be. If None, will be set by the Trainer.
-        worker_to_log (int): Worker index to log. By default, will log the
-            worker with index 0.
-    """
 
-    def __init__(self, logdir: Optional[str] = None,
-                 worker_to_log: int = 0) -> None:
-        self._logdir = logdir
-        self._workers_to_log = self._validate_worker_to_log(worker_to_log)
-        self._log_path = None
-
-    def _validate_worker_to_log(self, worker_to_log) -> int:
-        if isinstance(worker_to_log, Iterable):
-            worker_to_log = list(worker_to_log)
-            if len(worker_to_log) > 1:
-                raise ValueError(
-                    f"{self.__class__.__name__} only supports logging "
-                    "from a single worker.")
-            elif len(worker_to_log) < 1:
-                raise ValueError(
-                    "At least one worker must be specified in workers_to_log.")
-            worker_to_log = worker_to_log[0]
-        if not isinstance(worker_to_log, int):
-            raise TypeError("workers_to_log must be an integer.")
-        return worker_to_log
-
-
-class MLflowLoggerCallback(TrainingSingleWorkerLoggingCallback):
+class MLflowLoggerCallback(TrainingLogdirCallback, TrainingCallback):
```

Comment:
Now that this is no longer inheriting from `TrainingSingleWorkerLoggingCallback`, I don't think will work unless a `self._logdir = logdir` line is added to the `__init__`.